File: packages/zynth-templates/native-module/android/Zynth{{MODULE_NAME_PASCAL}}/src/main/java/dev/zynth/{{MODULE_NAME}}/{{MODULE_NAME_PASCAL}}Module.kt
package dev.zynth.{{MODULE_NAME}}

import com.zynth.kit.runtime.ZynthModule

import com.zynth.kit.runtime.ZynthSyncModule

import com.zynth.kit.runtime.ZynthArgs

import org.json.JSONObject



class {{MODULE_NAME_PASCAL}}Module : ZynthModule, ZynthSyncModule {

    override val name = "{{MODULE_NAME_PASCAL}}"



    override val constants: Map<String, Any>?

        get() = mapOf("exampleConstant" to "Hello from Android")



    override fun initialize() {

        // Module setup

    }



    override fun invalidate() {

        // Cleanup

    }



    override fun call(method: String, args: ZynthArgs): JSONObject {

        return when (method) {

            "exampleMethod" -> JSONObject().put("result", "Async result")

            else -> JSONObject().put("error", "unknown_method")

        }

    }



    override fun callSync(method: String, args: ZynthArgs): Any? {

        return when (method) {

            "exampleSyncMethod" -> "Sync result"

            else -> null

        }

    }

}

File: packages/zynth-templates/native-module/ios/{{MODULE_NAME_PASCAL}}Module.swift
import Foundation
import ZynthKit

@objc({{MODULE_NAME_PASCAL}}Module)
class {{MODULE_NAME_PASCAL}}Module: NSObject, ZynthModule, ZynthSyncModule {
  let name = "{{MODULE_NAME_PASCAL}}"
  
  var constantsToExport: [String: Any]? {
    ["exampleConstant": "Hello from iOS"]
  }

  func initialize() {
    // Module setup
  }

  func invalidate() {
    // Cleanup
  }

  func call(method: String, args: ZynthArgs) throws -> Any? {
    switch method {
    case "exampleMethod":
      return ["result": "Async result"]
    default:
      return nil
    }
  }
  
  func callSync(method: String, args: ZynthArgs) throws -> Any? {
    switch method {
    case "exampleSyncMethod":
      return "Sync result"
    default:
      throw ZynthModuleError.syncNotSupported(module: name, method: method)
    }
  }

  @objc public static func initialize(with runtime: ZynthRuntime) {
    let module = {{MODULE_NAME_PASCAL}}Module()
    runtime.installModules([module])
  }
}